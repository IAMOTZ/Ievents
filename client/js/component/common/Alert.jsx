import React from 'react';

const SmallAlert = (props) => {
  const component = props.message ? (
    <span className="text-danger text-center small">{props.message}</span>
  ) : null;
  return component;
};

const BigAlert = (props) => {
  const component = props.message ? (
    <div
      className={`alert text-center ${props.type === 'success' ? 'alert-success' : 'alert-danger'}`}
      role="alert"
    >{props.message}
    </div>
  ) : null;
  return component;
};

export { BigAlert, SmallAlert };
