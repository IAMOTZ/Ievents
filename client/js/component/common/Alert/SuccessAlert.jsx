import React from 'react';

export default (props) => {
  if (props.message) {
    return (
      <div className="alert alert-success" role="alert">
        <strong className="text-black">{props.message}</strong>
      </div>
    )
  }
  else {
    return null;
  }
};
