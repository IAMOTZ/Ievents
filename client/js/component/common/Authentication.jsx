import React from 'react';
import { Redirect } from 'react-router-dom';

export default (props) => {
  if(props.condition) {
    return null
  } else {
    return (<Redirect to="/" />);
  }
}