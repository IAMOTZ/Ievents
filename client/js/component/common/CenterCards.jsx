import React from 'react';
import { Link } from 'react-router-dom';

const defaultImage = '../../../images/defaultImgx4.jpeg';

const CenterCards = (props) => {
  return props.centers.map((center) => (
    <div className="col-lg-4 col-md-6 col-sm-12" key={center.id}>
      <div className="card">
        <div>
          <img className="card-img-top" src={center.images ? center.images[0] : defaultImage} alt="" />
          <span className="badge text-white p-2 seat-badge ">{center.capacity} seats</span>
        </div>
        <div className="card-body d-flex flex-column">
          <div>
            <h4 className="card-title text-capitalize">{center.name}</h4>
            <i className="fa fa-map-marker fa-fw" aria-hidden="true"></i>&nbsp; <span className="text-capitalize">{center.location}</span>
          </div>
          <ActionBtn isAdmin={props.isAdmin || false}
            id={center.id}
            editAction={props.editAction}
            btnAction={props.btnAction} />
        </div>
      </div>
    </div>
  ));
}

const ActionBtn = (props) => {
  if (props.isAdmin) {
    return (
      <Link to="/editCenter"
        class="btn text-white mt-3"
        id={props.id}
        onClick={props.editAction}>Edit</Link>
    );
  } else {
    return (
      <a role="button"
        class="btn text-white mt-3"
        data-toggle="modal"
        data-target="#center-details-modal"
        id={props.id}
        onClick={props.btnAction}>Details</a>
    );
  }
}

export default CenterCards;
