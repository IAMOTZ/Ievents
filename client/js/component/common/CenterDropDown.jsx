import React from 'react';

export default (props) => {
  const centerOptions = props.centers.map((center) => {
    if (center.id === props.selected) {
      return (<option value={center.id} key={center.id} name="centerId" selected>{center.name}</option>)
    } else {
      return (<option value={center.id} key={center.id} name="centerId">{center.name}</option>)
    }
  });
  return centerOptions;
}