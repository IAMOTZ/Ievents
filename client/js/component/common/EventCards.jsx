import React from 'react';

export default (props) => {
  const eventCards = props.events.map((event => {
    return (
      <div className="card text-dark bg-white mb-3 mx-auto" style={{ maxWidth: 20 + 'rem' }} key={event.id}>
        <div className="card-header h4">
          {event.title}
          <a href="" className="ml-4 mr-2">
            <i className="fa fa-pencil"></i>
          </a>
          <a href="#" className="text-primary">
            <i className="fa fa-trash" onClick={props.startDelete} id={event.id}></i>
          </a>
        </div>
        <div className="card-body">
          <p className="card-text">{event.description}</p>
          <span className="text-muted">{event.date}&nbsp;</span>
          <a href="#" className="btn btn-block bg-dark text-white mt-2" onClick={props.remove} id={event.id}>Done</a>
        </div>
      </div>
    );
  }));
  return eventCards;

}