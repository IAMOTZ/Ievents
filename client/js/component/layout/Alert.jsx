import React from 'react';
import { connect } from 'react-redux';

@connect((store) => {
  return {
    user: store.user
  }
})

export default class Alert extends React.Component {
  render() {
    if(this.props.user.status.error){
      return (
        <div className="alert alert-warning" role="alert">
          <strong className="text-black">{this.props.user.status.error.message}</strong>
        </div>
      )
    }
    else {
      return null;  
    }
  }
}