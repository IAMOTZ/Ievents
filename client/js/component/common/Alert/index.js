import SuccessAlert from './SuccessAlert.jsx';
import WarningAlert from './WarningAlert.jsx';

export { SuccessAlert, WarningAlert };
