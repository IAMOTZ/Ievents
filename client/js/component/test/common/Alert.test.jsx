/* global shallow */
import React from 'react';
import { BigAlert, SmallAlert } from '../../common/Alert';

const props = {
  message: null,
  type: null,
};

const alterProps = newProps => ({
  ...props, ...newProps,
});

describe('<SmallAlert />', () => {
  it('should render nothing when there is no message', () => {
    const wrapper = shallow(<SmallAlert {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render correctly when there is message', () => {
    const wrapper = shallow(<SmallAlert {...alterProps({ message: 'Just an error' })} />);
    expect(wrapper).toMatchSnapshot();
  });
});
describe('<BigAlert />', () => {
  it('should render nothing when there is no message', () => {
    const wrapper = shallow(<BigAlert {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render correctly when there is message', () => {
    const wrapper = shallow(<BigAlert {...alterProps({ message: 'Just an error' })} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render a succes alert when type is set to success', () => {
    const wrapper = shallow(<BigAlert {...alterProps({ message: 'I am success', type: 'success' })} />);
    expect(wrapper).toMatchSnapshot();
  });
});
