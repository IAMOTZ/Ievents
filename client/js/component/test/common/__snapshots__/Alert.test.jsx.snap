// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BigAlert /> should render a succes alert when type is set to success 1`] = `
<div
  className="alert text-center alert-success"
  role="alert"
>
  I am success
</div>
`;

exports[`<BigAlert /> should render correctly when there is message 1`] = `
<div
  className="alert text-center alert-danger"
  role="alert"
>
  Just an error
</div>
`;

exports[`<BigAlert /> should render nothing when there is no message 1`] = `""`;

exports[`<SmallAlert /> should render correctly when there is message 1`] = `
<span
  className="text-danger text-center small"
>
  Just an error
</span>
`;

exports[`<SmallAlert /> should render nothing when there is no message 1`] = `""`;
